dataregistry-extension/src/explorer.tsx

```diff
@@ -166,12 +168,17 @@ export function DatasetCompononent({
         if (active === url) {
           classNames.push(activeDatasetClassName);
         }
+
+        const dataset = registry.getURL(url);
         // Use label if it exists
-        let label = labelDataType.getDataset(registry.getURL(url));
+        let label = labelDataType.getDataset(dataset);
+        const displayURL = externalURLDataType.getDataset(dataset) || of(url);
         // otherwise use url
         if (!label) {
-          label = of(
-            url.startsWith(parentURL) ? url.slice(parentURL.length) : url
+          label = combineLatest(displayURL, parentURL).pipe(
+            map(([u, parentU]) =>
```

Comment:
A very minor (pedantic) quibble. Generally, best to define functions at the highest possible scope. Why? Because otherwise we must rely on runtime compilers hoisting function definitions based on whether they need access to higher scopes. Here, the function body only needs access to the contents of the provided array argument, which means it does not need access to anything in the `DatasetComponent` scope.

So what might this look like? A small utility function given some sort of descriptive name defined within the top-level module scope.

Historically, one can ensure better performance by elevating function definitions as high as possible. In fact, unrelated to hoisting, V8 once chose to de-optimize (or rather punt on optimization altogether) if a function body exceeded `600` characters (including comments). However, as compilers continue to evolve, YMMV.

Aside: there is obviously a "readability" argument which can be made in support of "inlining" a function definition; however, we should be aware that there may exist a runtime performance trade-off.

Reply:
Thanks for the suggestion and explanation. The history there is pretty interesting. I always wonder how much of a difference micro performance things like this make on overall UX.

Reply:
In a former life when I was doing more server-side development with Node.js, one memorable instance of a performance bottleneck was precisely this issue: deeply nested closures. At the time, the difference was something like `100x`, but this has undoubtedly changed.